crates/ruff/src/codes.rs

```diff
@@ -26,691 +26,700 @@ impl PartialEq<&str> for NoqaCode {
-        (Pycodestyle, "E116") => Rule::UnexpectedIndentationComment,
-        (Pycodestyle, "E117") => Rule::OverIndented,
-        (Pycodestyle, "E201") => Rule::WhitespaceAfterOpenBracket,
-        (Pycodestyle, "E202") => Rule::WhitespaceBeforeCloseBracket,
-        (Pycodestyle, "E203") => Rule::WhitespaceBeforePunctuation,
-        (Pycodestyle, "E211") => Rule::WhitespaceBeforeParameters,
-        (Pycodestyle, "E221") => Rule::MultipleSpacesBeforeOperator,
-        (Pycodestyle, "E222") => Rule::MultipleSpacesAfterOperator,
-        (Pycodestyle, "E223") => Rule::TabBeforeOperator,
-        (Pycodestyle, "E224") => Rule::TabAfterOperator,
-        (Pycodestyle, "E225") => Rule::MissingWhitespaceAroundOperator,
-        (Pycodestyle, "E226") => Rule::MissingWhitespaceAroundArithmeticOperator,
-        (Pycodestyle, "E227") => Rule::MissingWhitespaceAroundBitwiseOrShiftOperator,
-        (Pycodestyle, "E228") => Rule::MissingWhitespaceAroundModuloOperator,
-        (Pycodestyle, "E231") => Rule::MissingWhitespace,
-        (Pycodestyle, "E251") => Rule::UnexpectedSpacesAroundKeywordParameterEquals,
-        (Pycodestyle, "E252") => Rule::MissingWhitespaceAroundParameterEquals,
-        (Pycodestyle, "E261") => Rule::TooFewSpacesBeforeInlineComment,
-        (Pycodestyle, "E262") => Rule::NoSpaceAfterInlineComment,
-        (Pycodestyle, "E265") => Rule::NoSpaceAfterBlockComment,
-        (Pycodestyle, "E266") => Rule::MultipleLeadingHashesForBlockComment,
-        (Pycodestyle, "E271") => Rule::MultipleSpacesAfterKeyword,
-        (Pycodestyle, "E272") => Rule::MultipleSpacesBeforeKeyword,
-        (Pycodestyle, "E273") => Rule::TabAfterKeyword,
-        (Pycodestyle, "E274") => Rule::TabBeforeKeyword,
-        (Pycodestyle, "E275") => Rule::MissingWhitespaceAfterKeyword,
-        (Pycodestyle, "E401") => Rule::MultipleImportsOnOneLine,
-        (Pycodestyle, "E402") => Rule::ModuleImportNotAtTopOfFile,
-        (Pycodestyle, "E501") => Rule::LineTooLong,
-        (Pycodestyle, "E701") => Rule::MultipleStatementsOnOneLineColon,
-        (Pycodestyle, "E702") => Rule::MultipleStatementsOnOneLineSemicolon,
-        (Pycodestyle, "E703") => Rule::UselessSemicolon,
-        (Pycodestyle, "E711") => Rule::NoneComparison,
-        (Pycodestyle, "E712") => Rule::TrueFalseComparison,
-        (Pycodestyle, "E713") => Rule::NotInTest,
-        (Pycodestyle, "E714") => Rule::NotIsTest,
-        (Pycodestyle, "E721") => Rule::TypeComparison,
-        (Pycodestyle, "E722") => Rule::BareExcept,
-        (Pycodestyle, "E731") => Rule::LambdaAssignment,
-        (Pycodestyle, "E741") => Rule::AmbiguousVariableName,
-        (Pycodestyle, "E742") => Rule::AmbiguousClassName,
-        (Pycodestyle, "E743") => Rule::AmbiguousFunctionName,
-        (Pycodestyle, "E902") => Rule::IOError,
-        (Pycodestyle, "E999") => Rule::SyntaxError,
+        (Pycodestyle, "E101") => (RuleGroup::Unspecified, Rule::MixedSpacesAndTabs),
```

Comment:
From a conceptual point. How is a `RuleGroup` different from the "Linter" (?) Group? 

E.g. `Pycodestyle` vs `Unspecified`. Could `Nursery` be another variant in the linter group?

Reply:
I can try it. My initial thinking was that it wouldn't be possible. The Linter encodes a prefix, so if you look at the bugbear rules as an example, they're defined like:

```
(Flake8Bugbear, "002") => Rule::UnaryPrefixIncrement,
```

This rule is selectable as `B002`, despite `B` not appearing in the definition, because `B` is associated with `Flake8Bugbear` rather than the rule. So on the surface, there'd be no way to use a `Nursery` group and associate rules with (e.g.) `E225`.

However, maybe it does work somehow? Pycodestyle is associated with two prefixes:

```
#[prefix = "E"]
#[prefix = "W"]
Pycodestyle,
```

I don't really know how that works. I'll look into it.

Reply:
Mmm, thinking on it more, there are other benefits to keeping these as separate concepts. For example, with the current setup, these rules will appear under the Pycodestyle section in the docs, which I think is good?

Reply:
That makes sense. Don't spend too much time on it. I now understand the reasoning behind it. Thank you.

Reply:
Do you think `RuleGroup` should instead be defined as part of the violation (ideally -- it's actually not possible to do with the proc macro, but ideally)? Alternatively, should we nix `RuleGroup` and _just_ do a nursery-oriented design for now?

Reply:
I'm not very familiar with our setup yet. I would find it ideal if there's a single place where I have to look for and define metadata for a rule. Whether this is on the rule or in the violation seems less important to me. 

Can you explain what you mean by a nursery-oriented design?

Reply:
By “nursery-oriented design”, I mean an approach that models the data as “nursery or not” (eg it could be a Boolean) rather than as a generic “rule group” concept and enumeration.

Reply:
I don't think I have a preference. A `bool` `RuleGroup` seems very similar to a two-variant enum `RuleGroup` to me. Let's not overthink this now.